### PR TITLE
jp-document-upload changes

### PR DESCRIPTION
**Pull request by 4p00rv**

### CODE REVIEW

File: src/javascript/app/pages/user/account/authenticate.js [hunk 1]
```diff
@@ -20,13 +20,19 @@ const Authenticate = (() => {
                     if (!/authenticated/.test(status)) {
                         init();
                         $('#not_authenticated').setVisibility(1);
-                        let link = 'https://marketing.binary.com/authentication/2017_Authentication_Process.pdf';
-                        if (Client.isAccountOfType('financial')) {
-                            $('#not_authenticated_financial').setVisibility(1);
-                            link = 'https://marketing.binary.com/authentication/2017_MF_Authentication_Process.pdf';
+                        // Show upload instructions
+                        if (jpClient()) {
+                            $('.jp-show').setVisibility(1);
```
Comment (negarn) [inline src/javascript/app/pages/user/account/authenticate.js]: I think you can use existing classes `ja-hide` and `ja-show`, js code already handles hiding and showing those

Comment (4p00rv) [reply src/javascript/app/pages/user/account/authenticate.js]: @negarn Doesn't seem to work.

File: src/javascript/app/pages/user/account/authenticate.js [hunk 2]
```diff
@@ -13,16 +13,27 @@ const Authenticate = (() => {
...
-                const status = response.get_account_status.status;
-                if (!/authenticated/.test(status)) {
-                    init();
-                    const $not_authenticated = $('#not_authenticated').setVisibility(1);
-                    let link = 'https://marketing.binary.com/authentication/2017_Authentication_Process.pdf';
-                    if (Client.isAccountOfType('financial')) {
-                        $('#not_authenticated_financial').setVisibility(1);
-                        link = 'https://marketing.binary.com/authentication/2017_MF_Authentication_Process.pdf';
+                const get_account_status  = response.get_account_status;
+                const should_authenticate = +get_account_status.prompt_client_to_authenticate;
```
Comment (binary-hamedanchi) [inline src/javascript/app/pages/user/account/authenticate.js]: In order to allow unauthenticated clients to upload their documents (i.e. for MT5 withdrawal), we are [not checking this flag](https://github.com/binary-com/binary-static/pull/3417/commits/f76609175865c5fd21d8d21572b28bd678d442d5#diff-fba57d9cce18f8e420f158772b4557e5L17) here anymore.

File: src/javascript/app/pages/user/account/authenticate.js [hunk 3]
```diff
@@ -13,16 +13,27 @@ const Authenticate = (() => {
...
+                            let link = 'https://marketing.binary.com/authentication/2017_Authentication_Process.pdf';
+                            if (Client.isAccountOfType('financial')) {
+                                $('#not_authenticated_financial').setVisibility(1);
+                                link = 'https://marketing.binary.com/authentication/2017_MF_Authentication_Process.pdf';
+                            }
+                            $not_authenticated.find('.learn_more a').attr('href', link);
+                        }
+
+                    } else if (!/age_verification/.test(status)) {
+                        $('#needs_age_verification').setVisibility(1);
```
Comment (binary-hamedanchi) [inline src/javascript/app/pages/user/account/authenticate.js]: To fix according to the previous comment, will also fix the duplication here with the next lines.

File: src/javascript/app/pages/user/account/authenticate.js [hunk 4]
```diff
@@ -221,45 +216,97 @@ const Authenticate = (() => {
...
+                localize('File ([_1]) size exceeds the permitted limit. Maximum allowed file size: 3MB', [file_checks[file.id].label]),
+                localize('ID number is required for [_1].', [file_checks[file.id].label]),
+                localize('Only letters, numbers, space, underscore, and hyphen are allowed for ID number ([_1]).', [file_checks[file.id].label]),
+                localize('Expiry date is required for [_1].', [file_checks[file.id].label]),
+                localize('Front and reverse side photos of [_1] are required.', [file_checks[file.id].label]),
+                // Japan Error message
+                localize('My number card is required.'),
+                localize('Please select at least one file from section 1 or two different file types from section 2.'),
+                localize('One of [_1] or [_2] is required.', [localize('Medical insurance card'), localize('Pension book')]),
+                localize('One of [_1], [_2] or [_3] is required.', [localize('Resident records'),
```
Comment (binary-hamedanchi) [inline src/javascript/app/pages/user/account/authenticate.js]: Missing the first string from `generate-static-data.js`. (same for the above line)

**Pull request closed without merging**